tests/FetchAuthTokenTest.php

```diff
@@ -53,10 +54,20 @@ class_implements($fetcherClass)
         )) {
             $mockFetcher->getQuotaProject()->shouldBeCalledTimes(1);
         }
-        $mockFetcher->fetchAuthToken(Argument::any())
-            ->shouldBeCalledTimes(1)
-            ->will($httpHandler);
+
+        if (is_a($fetcherClass, UpdateMetadataInterface::class, true)) {
+            $mockFetcher->updateMetadata(Argument::cetera())
+            ->shouldBeCalledTimes(1)->will(function () use (&$httpHandlerCalled) {
+                $httpHandlerCalled = true;
+                return ['authorization' => ['Bearer xyz']];
+            });
```

Comment:
nit: this should be indented once more
```suggestion
                ->shouldBeCalledTimes(1)->will(function () use (&$httpHandlerCalled) {
                    $httpHandlerCalled = true;
                    return ['authorization' => ['Bearer xyz']];
                });
```